qiskit/providers/ibmq/managed/ibmqjobmanager.py

```diff
@@ -0,0 +1,196 @@
+
+"""Job manager used to manage jobs for IBM Q Experience."""
+
+import logging
+from typing import List, Optional, Union
+from concurrent import futures
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import ScheduleComponent, Schedule
+from qiskit.compiler import transpile
+
+from .exceptions import IBMQJobManagerInvalidStateError
+from .utils import format_job_details, format_status_counts
+from .managedjobset import ManagedJobSet, RetrievedManagedJobSet
+from ..ibmqbackend import IBMQBackend
+from ..accountprovider import AccountProvider
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQJobManager:
+    """Job manager for IBM Q Experience."""
+
+    def __init__(self) -> None:
+        """Creates a new IBMQJobManager instance."""
+        self._job_sets = []  # type: List[ManagedJobSet]
+        self._executor = futures.ThreadPoolExecutor()
+
+    def run(
+            self,
+            experiments: Union[List[QuantumCircuit], List[Schedule]],
+            backend: IBMQBackend,
+            name: Optional[str] = None,
+            shots: int = 1024,
+            skip_transpile: bool = False,
+            max_experiments_per_job: Optional[int] = None
+    ) -> ManagedJobSet:
+        """Execute a set of circuits or pulse schedules on a backend.
+
+        The circuits or schedules will be split into multiple jobs. Circuits
+        or schedules in a job will be executed together in each shot.
+
+        The circuits will first be transpiled before submitted, using the
+        default transpiler settings. Specify ``skip_transpiler=True`` if you
+        want to skip this step.
+
+        A name can be assigned to this job set. Each job in this set will have
+        a job name consists of the set name followed by an underscore (_)
+        followed by the job index and another underscore. For example, a job
+        for set ``foo`` can have a job name of ``foo_1_``.  The name can then
+        be used to retrieve the jobs later.
+
+        Args:
+            experiments : Circuit(s) or pulse schedule(s) to execute.
+            backend: Backend to execute the experiments on.
+            name: Name for this set of jobs.
+            shots: Number of repetitions of each experiment, for sampling. Default: 1024.
+            skip_transpile: True if transpilation is to be skipped, otherwise False.
+            max_experiments_per_job: Maximum number of experiments to run in each job.
+                If not specified, the default is to use the maximum allowed by
+                the backend.
+                If the specified value is greater the maximum allowed by the
+                backend, the default is used.
+
+        Returns:
+            Managed job set.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the backend does not support
+                the experiment type.
+        """
+        if (any(isinstance(exp, ScheduleComponent) for exp in experiments) and
```

Comment:
I think the check here should be for `Schedule` and not `ScheduleComponent`, since the type hint says that the only acceptable inputs here are `QuantumCircuit` or `Schedule` -- so anything other than a `Schedule` should be caught by the hint. Here we are just trying to say that `Schedule` is supported by certain backends.